study.md

```diff
@@ -46,15 +51,46 @@ other named Mr.MagicNibbles (yellow) and bites a bit. Please copy the JSON you
 already wrote and include the information about Dave's fish.
 
 ```json
-<!-- your answer here -->
+{ "name":"Jason",
+  "things" : {
+    "cat": "Mr. Kitty",
+    "roommates": [
+      { "name": "Dave",
+        "things": {
+          "goldfish": [
+            { "name": "Bob", "color": "red", "bites": false},
+            { "name": "Mr. MagicNibbles", "color": "yellow", "bites": true}
+          ]
+        }
+      },
+      { "name": "Miller", "things": []}
+    ],
+    "loves": ["Red Sox", "Patriots", "Bruins"],
+    "recent-addresses": ["123 Fake Street", "2 Muchinfo Road"]
+  }
+}
 ```
 
 ## JSON Methods
 
 Using the JSON your wrote above, can you write a method on the JSON?
 
-```json
-<!-- your answer here -->
+```json`
+const petRoommatesGoldfish = function(json,roommateName,fishName) {
+  let roomies = json.things.roommates;
+  for ( let i = 0; i < roomies.length; i++ ) {
+    if (roomies[i].name === roommateName && roomies[i].things) {
+      let fishies = (roomies[i].things.goldfish || []);
+      for (let j = 0; j < fishies.length; j++) {
+        if (fishies[j].name === fishName) {
+          return (fishies[j].bites ? "YOW!!!" : "Aww, nice fishy.");
+        }
+      }
+      return `${roomies[i].name} doesn't have a fish by that name.`;
+    }
+  }
+  return "That roommate is imaginary.";
+};
```

Comment:
Yeah I sort of figured you'd need to do some sort of transform on the
string, but that once you did you could operate on it like a normal object.

     --Nick

On Mon, Mar 28, 2016 at 10:33 AM, Jason Weeks <notifications@github.com>
wrote:

> In study.md
> <https://github.com/ga-wdi-boston/json-study/pull/10#discussion_r57577312>
> :
>
> > +```json`
> > +const petRoommatesGoldfish = function(json,roommateName,fishName) {
> > +  let roomies = json.things.roommates;
> > +  for ( let i = 0; i < roomies.length; i++ ) {
> > +    if (roomies[i].name === roommateName && roomies[i].things) {
> > +      let fishies = (roomies[i].things.goldfish || []);
> > +      for (let j = 0; j < fishies.length; j++) {
> > +        if (fishies[j].name === fishName) {
> > +          return (fishies[j].bites ? "YOW!!!" : "Aww, nice fishy.");
> > +        }
> > +      }
> > +      return `${roomies[i].name} doesn't have a fish by that name.`;
> > +    }
> > +  }
> > +  return "That roommate is imaginary.";
> > +};
>
> nice. I see what you did. I was looking for a "no, you cant write methods
> in JSON" but this is golden. you would have to turn the JSON into an object
> however by doing a JSON.parse(jason)
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/ga-wdi-boston/json-study/pull/10/files/5d4b2b5f1c68e8d9bb9e053d938ce2ba96e1b13f#r57577312>
>